core/store/src/cold_storage.rs

```diff
@@ -136,6 +144,27 @@ pub fn update_cold_head<D: Database>(
     return Ok(());
 }
 
+/// Copies all contents of all cold columns from `hot_store` to `cold_db`.
+/// Does it column by column, and because columns can be huge, writes in batches.
+/// Threshold batch size is specified by `max_batch_size`.
+pub fn copy_all_data_to_cold<D: Database>(
+    cold_db: &ColdDB<D>,
+    hot_store: &Store,
+    max_batch_size: usize,
+) -> io::Result<()> {
+    for col in DBCol::iter() {
+        if col.is_cold() {
+            let mut transaction = BatchTransaction::new(max_batch_size);
+            for result in hot_store.iter(col) {
```

Comment:
> set the cold head to final head from before the copy operation was started

That's wrong, it should be "set the cold head to final head from before the copy":
```
final_head = get_final_head();
copy();
set_cold_head(final_head);
```

Reply:
Added fn to create checkpoint and an example of how I want to use to cold store tool. As such, shouldn't matter if we read head before or after copy because checkpoint db will not change during migration.

Reply:
This is old now.